src/main/java/com/ael/viner/util/MiningUtils.java

```diff
@@ -0,0 +1,157 @@
+package com.ael.viner.util;
+
+import com.ael.viner.registry.VinerBlockRegistry;
+import com.mojang.logging.LogUtils;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.tags.TagKey;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.enchantment.EnchantmentHelper;
+import net.minecraft.world.item.enchantment.Enchantments;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraftforge.registries.ForgeRegistries;
+import org.slf4j.Logger;
+
+import java.util.*;
+
+public class MiningUtils {
+
+    private static final Logger LOGGER = LogUtils.getLogger();
+
+    public static void mineBlocks(ServerPlayer player, List<BlockPos> blocksToMine) {
+        if (player == null)
+            return;
+
+        Level level = player.level();
+        ItemStack tool = player.getItemInHand(InteractionHand.MAIN_HAND);
+
+        // I don't think this is possible, but can't be sure.
+        if (level.isClientSide())
+            return;
+
+        // We ideally want all the drops to spawn at the first blockPos
+        BlockPos firstBlockPos = blocksToMine.get(0);
+
+        for (BlockPos blockPos: blocksToMine) {
+
+            // Getting block state of connected block
+            BlockState blockState = level.getBlockState(blockPos);
+
+            // Checking for Silk Touch enchantment
+            boolean hasSilkTouch = EnchantmentHelper.getTagEnchantmentLevel(Enchantments.SILK_TOUCH, tool) > 0;
+
+            if (hasSilkTouch) {
+                Block.popResource(level, firstBlockPos, new ItemStack(blockState.getBlock()));
+            } else {
+                // FIXME: We need to implement logic for Fortune
```

Comment:
Implemented a temp workaround for this yesterday by trying to replicate Vanilla fortune logic as close as possible